### PR TITLE
Remove deprecated and unused content areas from SearchBarComponent

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

**Comment by cbeer** [comment]
I'm not sure why you're removing `append` and `prepend`. I thought they would be useful for applications?

File: app/components/blacklight/metadata_field_layout_component.rb [hunk 1]
```diff
@@ -3,7 +3,8 @@
 module Blacklight
   class MetadataFieldLayoutComponent < ::ViewComponent::Base
     with_collection_parameter :field
-    with_content_areas :label, :value
+    renders_one :label
+    renders_one :value
```
Comment (cbeer) [inline app/components/blacklight/metadata_field_layout_component.rb:7]: I'm idly wondering if `renders_many :value` would be helpful, but i'm still trying to understand the new slots approach..

regardless, do we need to bump the view_component minimum version?

Comment (jcoyne) [reply app/components/blacklight/metadata_field_layout_component.rb:7]: Are they used by anything? I didn't see them used in Blacklight, and they don't have any tests or documentation that I could find, so I'm having trouble visualizing how they would be used.

**Comment by jcoyne** [comment]
Are pretend/append used by anything? I didn't see them used in Blacklight, and they don't have any tests or documentation that I could find, so I'm having trouble visualizing how they would be used.

**Comment by cbeer** [comment]
They can be used by downstream components. I don't think anything exists to do that, but it seems like an obvious thing that people are doing today by overriding the whole partial, so I don't know why we'd take that slot away.

I'm not sure what documentation you're expecting other than something that says... these content areas exist.. and to me, it feels like any test for them would just be exercising view_components for the sake of it, no?

**Comment by jcoyne** [comment]
superseded by #2443

**Pull request closed without merging**